src/averageAges.js

```diff
@@ -15,11 +15,25 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  return people
+    .filter(({ sex, died }) => {
+      const deathCentury = Math.ceil(died / 100);
+      const isCenturyRequired = century
+        ? sex === 'm' && deathCentury === century
+        : sex === 'm';
+
+      return isCenturyRequired;
+    })
+    .reduce((acc, person, index, array) => {
```

Comment:
You make the same calculations in the `calculateWomenAverageAge` function - it could be transferred to a separate variable and reused as a helper.

Reply:
done